File: exams/frmCreateNationalExam.vb
Public Class frmCreateNationalExam


    Private Sub create_examination()
        If qwrite("INSERT INTO national_examinations VALUES(NULL, '" & escape_string(Trim(txtExamName.Text).ToUpper) & "','" & cboYear.SelectedItem & "')") Then
            success("Examination Entry Successful!")
        Else
            failure("Examination Entry Failed!" & vbNewLine & "Duplicate Entry For Examination!")
        End If
    End Sub

    Private Function exists()
        If qread("SELECT * FROM national_examinations WHERE Name='" & escape_string(Trim(txtExamName.Text).ToUpper) & "' AND Year='" & cboYear.SelectedItem & "'") Then
            If dbreader.RecordsAffected > 0 Then
                dbreader.Close()
                Return True
            Else
                dbreader.Close()
                Return False
            End If
        Else
            failure("Could Not Read From The National Examinations Database!")
            Return True
        End If
    End Function
    Private Sub btnCreateExam_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCreateExam.Click
        If Trim(txtExamName.Text) <> "" And Not exists() Then
            create_examination()
        Else
            failure("Invalid Value For Examination Name!")
        End If
    End Sub

    Private Sub frmCreateNationalExam_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Not connect() Then
            Me.Close()
        Else
            For k As Integer = startyear To endyear
                cboYear.Items.Add(k)
            Next
            cboYear.SelectedItem = Today.Year
        End If
    End Sub
End Class

File: exams/frmCreateNationalExam.cs
using System;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace exams
{
    public partial class frmCreateNationalExam
    {
        public frmCreateNationalExam()
        {
            InitializeComponent();
            _btnCreateExam.Name = "btnCreateExam";
        }

        private void create_examination()
        {
            if (publicSubsNFunctions.qwrite(Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject("INSERT INTO national_examinations VALUES(NULL, '" + publicSubsNFunctions.escape_string(Strings.Trim(txtExamName.Text).ToUpper()) + "','", cboYear.SelectedItem), "')"))))
            {
                publicSubsNFunctions.success("Examination Entry Successful!");
            }
            else
            {
                publicSubsNFunctions.failure("Examination Entry Failed!" + Constants.vbNewLine + "Duplicate Entry For Examination!");
            }
        }

        private object exists()
        {
            string argq = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject("SELECT * FROM national_examinations WHERE Name='" + publicSubsNFunctions.escape_string(Strings.Trim(txtExamName.Text).ToUpper()) + "' AND Year='", cboYear.SelectedItem), "'"));
            if (publicSubsNFunctions.qread(ref argq))
            {
                if (publicSubsNFunctions.dbreader.RecordsAffected > 0)
                {
                    publicSubsNFunctions.dbreader.Close();
                    return true;
                }
                else
                {
                    publicSubsNFunctions.dbreader.Close();
                    return false;
                }
            }
            else
            {
                publicSubsNFunctions.failure("Could Not Read From The National Examinations Database!");
                return true;
            }
        }

        private void btnCreateExam_Click(object sender, EventArgs e)
        {
            if (Conversions.ToBoolean(Operators.AndObject((Strings.Trim(txtExamName.Text) ?? "") != (string.Empty ?? ""), !exists())))
            {
                create_examination();
            }
            else
            {
                publicSubsNFunctions.failure("Invalid Value For Examination Name!");
            }
        }

        private void frmCreateNationalExam_Load(object sender, EventArgs e)
        {
            if (!publicSubsNFunctions.connect())
            {
                Close();
            }
            else
            {
                for (int k = publicSubsNFunctions.startyear, loopTo = publicSubsNFunctions.endyear; k <= loopTo; k++)
                    cboYear.Items.Add(k);
                cboYear.SelectedItem = DateAndTime.Today.Year;
            }
        }
    }
}